src/plugins/chart_expressions/expression_heatmap/public/components/heatmap_component.tsx

```diff
@@ -289,22 +289,23 @@ export const HeatmapComponent: FC<HeatmapRenderProps> = memo(
       let end = index === array.length - 1 ? Number.POSITIVE_INFINITY : array[index + 1];
       // if the lastRangeIsRightOpen is set to false, we need to set the last range to the max value
       if (args.lastRangeIsRightOpen === false) {
-        const lastBand = max === start ? Number.POSITIVE_INFINITY : endValue;
+        const lastBand = max === min ? Number.POSITIVE_INFINITY : endValue;
         end = index === array.length - 1 ? lastBand : array[index + 1];
       }
 
       let overwriteArrayIdx;
 
       if (end === Number.POSITIVE_INFINITY) {
-        overwriteArrayIdx = `≥ ${start}`;
+        const startValue = index === array.length - 1 && max === min ? min : start;
```

Comment:
nit: I think it can be moved after line 289: `let end = index === array.length -.....`
in this case we can avoid duplicating